### DISCUSSION TITLE
包源码调试的问题

### DISCUSSION THREAD

**Discussion by MasterBalll** [xmake-io/xmake#4797] [General]
安装包时，包源码会放在 cache 中，但这个目录每月清理，所以如何能长久地对一个包进行源码调试？

**Comment by waruqi** [comment]
可以直接自己解压包源码，进入目录后，`xrepo install -d sourcedir zlib` 指定源码目录安装，这种方式，还能随意改包源码调试。。

**Comment by MasterBalll** [comment]
这个能否做全局指定阿？因为有不同的包（自建库）要调试，而且每个包的配置选项还有差异，这样搞起来可能会很麻烦。
或者如果能清理缓存的时候，把上个月最新版本的源码拷贝过来也行 hh

**Comment by waruqi** [comment]
稍微改了下，`add_requires("xxx", {debug = true})` ，如果是 debug 包，会保留源码目录，不过跨月的，搞不了，你就自己重新安装编译下 ，一个月也就一次，不差这点

**Comment by MasterBalll** [comment]
刚看到……
这样应该也能满足需要了，至少过几个月调试的时候能找到源码，回头我试下
